### DISCUSSION TITLE
Changing theme name before build

### DISCUSSION THREAD

**Discussion by mikedefieslife** [gregsullivan/_tw#101] [Q&A]
This is probably a daft question, but how/where do we change the theme slug, name etc.. that we set during the initial generation on the underscore tailwind website? 

Obviously I can change the theme name for the theme in file-header.css, but I mean for the overall project to that when I run npm run bundle, I get the zip name I want, and script names I want etc..

**Comment by gregsullivan** [comment]
Hi Mike—Thanks for your question!

I don't think it's daft, as it's not documented or really expected as a pathway.

You're right that the first step is updating `file-header.css`.

Which script names specifically are you finding don't have the proper names?

In terms of the zip file itself, you'll find it named in your `package.json` file on this line:

```json
"zip": "node node_scripts/zip.js [your-theme-slug]",
```

Please let me know if that covers everything for you!

**Comment by mikedefieslife** [comment]
Ok so in my file-header.css I have the following:

/*!
Theme Name: My Theme name
Theme URI: my theme URI
Author: Me
Author URI: My website
Description: My description
Version: 0.1.0
Tested up to: 6.2
Requires PHP: 7.4
License: GNU General Public License v2 or later
License URI: LICENSE
Text Domain: My txt_domain
Tags:

*/

When I build the theme the outputted file is na.zip. The theme folder is called na, and of course the links in the theme files things like /na/functions.php etc..

I presume when entering details on the underscore tailwind website, I either missed a field, or wrote NA (as in not applicable). Now I can't find where I can change the theme slug.

**Reply by gregsullivan** [reply]
This is how you fix the zip:

> In terms of the zip file itself, you'll find it named in your package.json file on this line:
> 
> ```json
> "zip": "node node_scripts/zip.js [your-theme-slug]",
> ```

If you want to fix all of the function names and the text domain, you'll need to use some clever search-and-replace.

If you haven't done much work on the theme, it might be easier just to generate a new one.

In `file-header.css`, you only really need to change the theme name, and only if it's not already what you want. If the text domain bothers you, you can change it as well, but you'll need to change every instance of it in your theme alongside the mention in `file-headers.css`.

**Comment by mikedefieslife** [comment]
Ok, I think what I'm wondering is, where is the 'theme slug' being read from? The value must be surely be stored somewhere for use at the time of the bundle. If it's stored somewhere, then it stands to reason changing that value would achieve what I want.

**Reply by gregsullivan** [reply]
It's not being stored in a single place. This is oversimplified, but here is how _s describes a manual version of the process to swap the slug:

> Clone or download this repository, change its name to something else (like, say, `megatherium-is-awesome`), and then you'll need to do a six-step find and replace on the name in all the templates.
> 
> 1. Search for `'_s'` (inside single quotations) to capture the text domain and replace with: `'megatherium-is-awesome'`.
> 2. Search for `_s_` to capture all the functions names and replace with: `megatherium_is_awesome_`.
> 3. Search for `Text Domain: _s` in `style.css` and replace with: `Text Domain: megatherium-is-awesome`.
> 4. Search for <code>&nbsp;_s</code> (with a space before it) to capture DocBlocks and replace with: <code>&nbsp;Megatherium_is_Awesome</code>.
> 5. Search for `_s-` to capture prefixed handles and replace with: `megatherium-is-awesome-`.
> 6. Search for `_S_` (in uppercase) to capture constants and replace with: `MEGATHERIUM_IS_AWESOME_`.
> 
> Then, update the stylesheet header in `style.css`, the links in `footer.php` with your own information and rename `_s.pot` from `languages` folder to use the theme's slug. Next, update or delete this readme.

The generator is doing this process automatically, with some other changes included.

The theme slug is never "read"; it exists in the theme in many contexts for a range of purposes. If you're interested in changing it in a specific instance, I'd be happy to help with that instance!
